internal/commands/vaultsecrets/run/run.go

```diff
@@ -139,20 +139,17 @@ func runRun(opts *RunOpts) (err error) {
 }
 
 func getAllSecretsForEnv(opts *RunOpts) ([]string, error) {
-	params := preview_secret_service.NewOpenAppSecretsParamsWithContext(opts.Ctx)
-	params.OrganizationID = opts.Profile.OrganizationID
-	params.ProjectID = opts.Profile.ProjectID
-	params.AppName = opts.AppName
-
-	res, err := opts.PreviewClient.OpenAppSecrets(params, nil)
+	secs, err := fetchPaginatedSecrets(opts)
 	if err != nil {
 		return nil, err
+	} else if len(secs) == 0 {
+		return nil, fmt.Errorf("no secrets returned from app")
```

Comment:
I think this is a reasonable UX improvement but happy to remove if folks don't think its needed

Reply:
Is there a story where you'd want to "run" even if the list is empty?

Reply:
Yeah great question. I suppose this is a breaking change and probably best removed.

I can think of a situation where you push app code that no longer needs secrets and you delete them from HVS but your tooling is still launching the app via `hcp vs run`

Reply:
Removed in 526b17d , we can add later if we want it.